src/sage/schemes/elliptic_curves/hom_composite.py

```diff
@@ -120,33 +118,98 @@ def _eval_factored_isogeny(phis, P):
+    wide range of situations.
+
     EXAMPLES::
 
         sage: from sage.schemes.elliptic_curves import hom_composite
         sage: E = EllipticCurve(GF(8191), [1,0])                                        # optional - sage.rings.finite_rings
         sage: P = E.random_point()                                                      # optional - sage.rings.finite_rings
-        sage: (l,e), = P.order().factor()                                               # optional - sage.rings.finite_rings
-        sage: phis = hom_composite._compute_factored_isogeny_prime_power(P, l, e)       # optional - sage.rings.finite_rings
+        sage: (l,n), = P.order().factor()                                               # optional - sage.rings.finite_rings
+        sage: phis = hom_composite._compute_factored_isogeny_prime_power(P, l, n)       # optional - sage.rings.finite_rings
         sage: hom_composite._eval_factored_isogeny(phis, P)                             # optional - sage.rings.finite_rings
         (0 : 1 : 0)
-        sage: [phi.degree() for phi in phis] == [l]*e                                   # optional - sage.rings.finite_rings
+        sage: [phi.degree() for phi in phis] == [l]*n                                   # optional - sage.rings.finite_rings
+        True
+
+    All choices of ``split`` produce the same result, albeit
+    not equally fast::
+
+        sage: E = EllipticCurve(GF(2^127 - 1), [1,0])                                           # optional - sage.rings.finite_rings
+        sage: P, = E.gens()                                                                     # optional - sage.rings.finite_rings
+        sage: (l,n), = P.order().factor()                                                       # optional - sage.rings.finite_rings
+        sage: phis = hom_composite._compute_factored_isogeny_prime_power(P,l,n)                 # optional - sage.rings.finite_rings
+        sage: phis == hom_composite._compute_factored_isogeny_prime_power(P,l,n, split=0)       # optional - sage.rings.finite_rings
+        True
+        sage: phis == hom_composite._compute_factored_isogeny_prime_power(P,l,n, split=0.1)     # optional - sage.rings.finite_rings
+        True
+        sage: phis == hom_composite._compute_factored_isogeny_prime_power(P,l,n, split=0.5)     # optional - sage.rings.finite_rings
+        True
+        sage: phis == hom_composite._compute_factored_isogeny_prime_power(P,l,n, split=0.9)     # optional - sage.rings.finite_rings
+        True
+        sage: phis == hom_composite._compute_factored_isogeny_prime_power(P,l,n, split=1)       # optional - sage.rings.finite_rings
         True
     """
-    E = P.curve()
-    phis = []
-    for i in range(e):
-        K = l**(e-1-i) * P
-        phi = EllipticCurveIsogeny(E, K)
-        E = phi.codomain()
-        P = phi(P)
-        phis.append(phi)
-    return phis
+    def rec(Q, k):
+        assert k
+        if k == 1:  # base case
+#            assert Q and not l*Q
```

Comment:
L196, L198 may be removed. They seem to be remnants from the testing period.

Reply:
I think they are useful — see https://github.com/sagemath/sage/pull/35270#issuecomment-1469386941 for my reasoning.

Reply:
I don't get the reasoning. You may regard me as someone who read your code later for whatever reason. They were not useful to me at all when I first was reading the code to understand the algorithm. Now that I understand the algorithm, I don't see why they are useful. 

> I can replace them with textual descriptions of the expected behaviour of the code if you prefer

Yes. I prefer that.

Instead of `        assert k`, just write `# k should be a positive integer`.  If you need `assert k` to warn the user for bad input `n`, then I would recommend to add `INPUT: n -- a positive integer` instead.

Instead of `#            assert Q and not l*Q`, write `# Q is not zero but l*Q is`.

The textual description is clear and explicit. I needed some exercise of my mind to get the meaning of your "code comment".

Reply:
Okay, seems like we have different preferences. How do you feel about this version?

```diff
     def rec(Q, k):
-        assert k
-        if k == 1:  # base case
-#            assert Q and not l*Q
+
+        if k == 1:
+            # base case: Q has order l
             return [EllipticCurveIsogeny(Q.curve(), Q)]

+        # recursive case: k>1 and Q has order l^k
+
         k1 = int(k * split + .5)
         k1 = max(1, min(k-1, k1))  # clamp to [1, k-1]
```

Reply:
Good for me.

Would you also put spaces around `+` and `-` in comments and codes? Some people spend time to fix them in sage codebase.

Reply:
Done.